subprojects/docs/src/docs/userguide/tasks.xml

```diff
@@ -290,52 +290,673 @@
+                    </sample>
+                    <para>As you can see, we can both configure the tasks properties and use those
+                        properties as arguments to the incremental build runtime API. Using the runtime
+                        API like this is a little like using <literal>doLast()</literal> and
+                        <literal>doFirst()</literal> to attach extra actions to a task, except in this
+                        case we’re attaching information about inputs and outputs. Note that if the
+                        task type is already using the incremental build annotations, the runtime API
+                        will add inputs and outputs rather than replace them.
+                    </para>
+                </section>
+                
+                <section id="sec:runtime_api_configuration">
+                    <title>Fine-grained configuration</title>
+                    <para>The runtime API methods only allow you to declare your inputs and outputs in
+                        themselves. However, the file-oriented ones return a builder - of type
+                        <apilink class="org.gradle.api.tasks.TaskInputFilePropertyBuilder"/> - that let’s
+                        you provide additional information about those inputs and outputs.
+                    </para>
+                    <para>You can learn about all the options provided by the builder in its API
+                        documentation, but we’ll show you a simple example here to give you an idea of
+                        what you can do.
+                    </para>
+                    <para>Let’s say we don’t want to run the <literal>processTemplates()</literal> task
+                        if there are no source files, regardless of whether it’s a clean build or not.
+                        After all, if there are no source files, there’s nothing for the task to do. The
+                        builder allows us to configure this like so:
+                    </para>
+                    <sample id="runtimeIncrementalApiConfiguration"
+                            dir="userguide/tasks/incrementalBuild/customTaskClass"
+                            title="Using skipWhenEmpty() via the runtime API">
+                        <sourcefile file="build.gradle" snippet="runtime-api-conf"/>
+                        <output args="clean processTemplatesRuntimeConf"
+                                ignoreExtraLines="true"
+                                ignoreLineOrder="true"/>
+                    </sample>
+                    <para>So the <literal>TaskInputs.files()</literal> method returns a builder that has
+                        a <literal>skipWhenEmpty()</literal> method. By calling this method, we tell
+                        Gradle that the task should be considered up to date if the corresponding input
+                        file collection is empty, even if the task hasn't run before.
```

Comment:
Maybe it's worth mentioning that prior to 3.0 we had `TaskInputs.source()` and `sourceDir()` methods for this purpose. These are now deprecated in favor of using `files(...).skipWhenEmpty()`.